KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
+            "\\subsubsection*{" in line):
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 1:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line) and "orthogonality relation" not in line.lower():
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 2:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line):
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+    for i in khyper_header_chap:
+        if i == "Pseudo Jacobi (or Routh-Romanovski)":
+            khyper_header_chap[khyper_header_chap.index(i)] = "Pseudo Jacobi"
+    k_hyp_index_iii = 0
+    offset = 0
+
+    i = 0
+    while i < len(khyper_header_chap):
+        try:
+            if khyper_header_chap[i] == khyper_header_chap[i+1]:
+                k_hyper_sub_chap[i + 1] = k_hyper_sub_chap[i] + "\paragraph{\\bf KLS Addendum: " + \
+                    word + "}\n" + k_hyper_sub_chap[i + 1]
+                khyper_header_chap[i] = "memes"
+                k_hyper_sub_chap[i] = "memes"
+            else:
+                i += 1
+        except IndexError:
+            i += 1
+
+    a = 0
+    while a < len(khyper_header_chap):
+        if khyper_header_chap[a] == "memes":
+            del khyper_header_chap[a]
+            del k_hyper_sub_chap[a]
+        else:
+            a += 1
+
+    global sorterIIIcheck
+    chap9 = 0
+
+    if sorterIIIcheck[sortloc][0] == 0:
+        tempref = ref9_3
+        sorterIIIcheck[sortloc][0] += 1
+        chap9 = 1
+    else:
+        tempref = ref14_3
+        offset = sorterIIIcheck[sortloc][1]
+        if sep1 == 1:
+            offset = 8
+
+    for d in range(0, len(tempref)):  # check every section and subsection line
```

Comment:
When using `len`, the first argument is assumed to be 0 (omit it if it is just supposed to be 0).

Also, use `enumerate` instead of `range` and `len`.